tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/CodeownersTools/CodeownersTools.cs

```diff
@@ -0,0 +1,483 @@
+                }
+
+                // Find existing codeowners entry by path or service label
+                var matchingEntry = codeownersHelper.FindMatchingEntries(mergedCodeownersEntries, normalizedPath, serviceLabel);
+
+                var codeownersEntryExisted = false;
+                var updatedEntry = new CodeownersEntry();
+                if (matchingEntry != null)
+                {
+                    updatedEntry = codeownersHelper.UpdateCodeownersEntry(
+                        matchingEntry, normalizedPath, serviceLabel, serviceOwners, sourceOwners, isAdding, isMgmtPlane);
+                    codeownersEntryExisted = true;
+                }
+                else
+                {
+                    updatedEntry = codeownersHelper.CreateCodeownersEntry(
+                        normalizedPath, serviceLabel, serviceOwners, sourceOwners, isMgmtPlane
+                    );
+                }
+
+                // Validate the modified/created Entry
+                var (validationErrors, codeownersValidationResults) = await ValidateMinimumOwnerRequirements(updatedEntry);
+
+                var codeownersValidationResultMessage = string.Join("\n", codeownersValidationResults.Select(r => r.ToString()));
+                if (validationErrors.Any())
+                {
+                    throw new Exception($"{validationErrors} Validation results: {codeownersValidationResultMessage}");
+                }
+
+                // Modify the file
+                var modifiedCodeownersContent = codeownersHelper.AddCodeownersEntry(mergedCodeownersEntries, codeownersContent, updatedEntry, codeownersEntryExisted);
+
+                // Create Branch, Update File, and Handle PR.
+                var identifier = !string.IsNullOrWhiteSpace(updatedEntry.ServiceLabels?.FirstOrDefault())
+                    ? updatedEntry.ServiceLabels.FirstOrDefault()
+                    : updatedEntry.PathExpression;
+                var resultMessages = await CreateCodeownersPR(
+                    repo,                                                             // Repository name
+                    modifiedCodeownersContent,                     // Modified content
+                    codeownersSha,                                                    // SHA of the file to update 
+                    $"Update codeowners entry for {identifier}", // Description for commit message, PR title, and description
+                    "update-codeowners-entry",                                             // Branch prefix for the action
+                    identifier, // Identifier for the PR
+                    workingBranch);
+
+                return string.Join("\n", resultMessages.Concat(new[] { codeownersValidationResultMessage }));
+            }
+            catch (Exception ex)
+            {
+                SetFailure();
+                return $"Error: {ex}";
```

Comment:
I would do something like this, that way we get stack trace logging and a clean error message as the response.

```suggestion
                SetFailure();
                logger.LogError(ex);
                return ex.Message;
```